packages/playwright-core/src/server/chromium/crNetworkManager.ts

```diff
@@ -158,10 +175,23 @@ export class CRNetworkManager {
     });
   }
 
+  // Service Worker Network.requestWillBeSent event fires on the parent frame session,
+  // but the Fetch.requestPaused comes in on the Service Worker session.
+  // We assume these came in the the same session (on the Service Worker), so shuffle things around.
+  // In the future, this may be resolved; see https://crbug.com/1304536
+  _maybeAdoptServiceWorkerMainRequest(id: string) {
+    const event = this._parentManager?._requestIdToRequestWillBeSentEvent.get(id);
+    if (event) {
+      this._parentManager?._requestIdToRequestWillBeSentEvent.delete(id);
+      this._requestIdToRequestWillBeSentEvent.set(id, event);
+    }
+
+    return this._requestIdToRequest.get(id);
+  }
+
   _onRequestPaused(workerFrame: frames.Frame | undefined, event: Protocol.Fetch.requestPausedPayload) {
     if (!event.responseStatusCode && !event.responseErrorReason) {
-      // Request intercepted, deliver signal to the tracker.
-      const request = this._requestIdToRequest.get(event.networkId!);
+      const request = this._requestIdToRequest.get(event.networkId!) || this._maybeAdoptServiceWorkerMainRequest(event.networkId!);
```

Comment:
Add a test that `'request'` event is dispatched for the main sw script without interception, and is not dispatched on the page.